### ISSUE TITLE
Flycheck should ignore stderr output from checker

### ISSUE THREAD

**Issue by aifreedom** [flycheck/flycheck#1323]
This is triggered by an eslint bug https://github.com/airbnb/javascript/issues/1488. The error log is printed in stderr, but it was picked up by flycheck as part of the XML output of `eslint` thus the output is not valid XML blob and breaks flycheck.

The eslint plugin bug has been fixed in https://github.com/yannickcr/eslint-plugin-react/pull/1292 but before it is released, flycheck eslint is broken for me in Emacs. 

* Emacs version: `GNU Emacs 25.1.1 (x86_64-apple-darwin15.6.0, NS appkit-1404.47 Version 10.11.6 (Build 15G31)) of 2016-10-09`
* Flycheck version: `20170907.1504`
* eslint version `3.10.10`

Below is the error message I got in Emacs.

```
Suspicious state from syntax checker javascript-eslint: Flycheck checker javascript-eslint returned non-zero exit code 1, but its output contained no errors: can't resolve reference #/definitions/basicConfig from id #
can't resolve reference #/definitions/basicConfigOrBoolean from id #
can't resolve reference #/definitions/basicConfigOrBoolean from id #
<?xml version="1.0" encoding="utf-8"?><checkstyle version="4.3"><file name="/path/to/file.jsx"><error line="84" column="7" severity="error" message="&apos;variable&apos; is defined but never used. (no-unused-vars)" source="eslint.rules.no-unused-vars" /></file></checkstyle>

Try installing a more recent version of javascript-eslint, and please open a bug report if the issue persists in the latest release.  Thanks!
```

**Comment by fmdkdd** [comment]
To be clear: you are arguing that Flycheck should ignore stderr *for the eslint checker* (to avoid choking on stderr warnings) , not that it should *always ignore stderr*, regardless of the checker?

I agree that it is unfortunate that Flycheck fails to parse the errors in this case.  On the other hand, if `eslint` uses stderr to signal an internal failure, it seems to me more useful to report to the user than to ignore it.  What if, after a failure, `eslint` does not provide any lints?  stdout would be empty, and stderr would contain the failure message, but if Flycheck filters stderr out, you would think that everything is alright, while it really isn't.

If the incriminating plugin has been fixed, but you cannot use the fixed version, the workaround in your case would be to override the `eslint` binary in your `PATH` with your own shell script that calls the original `eslint`, redirects stderr to `/dev/null` and leaves stdout alone.  You could also customize `flycheck-executable-find` if you wanted the override to only happen for Flycheck.

cc @Simplify

**Comment by cpitclaudel** [comment]
Thanks for the report @aifreedom.  We tend to not use workarounds for checker bugs, especially not after they've been fixed, at is end up being just too much work with all the checkers.  Please feel free to reopen if there are strong reasons not to upgrade to the latest version, though.

**Comment by ibrahima** [comment]
Is there a way to ignore stderr by changing the checker definition? I've tried adding "2> /dev/null" to the command for a checker, but this actually gets passed as an argument to the checker binary, so it doesn't do what I want.

For context, I'm having an issue with slim-lint where the bundled ruby parser library complains that my ruby version doesn't match the version the parser is built for (parser supports 2.3.3 while I'm on 2.3.5). I'd like to be able to ignore this so that I can keep getting slim-lint feedback. Not sure of a way to fix this yet, although perhaps as a hack I can define my own wrapper for slim-lint that just redirects stderr to /dev/null.

**Comment by fmdkdd** [comment]
@ibrahima Your wrapper is the easiest workaround.  There are no facilities to ignore stderr on a per-checker basis in Flycheck itself, as they all go through the same `start-process` call.

**Comment by fmdkdd** [comment]
For anyone encountering the same issue in the first post, try updating `eslint-plugin-react`.  The fix has been released in version 7.2.
